pipenv/vendor/tomlkit/container.py

```diff
@@ -189,6 +189,10 @@ def remove(self, key):  # type: (Union[Key, str]) -> Container
         if idx is None:
             raise NonExistentKey(key)
 
+        old_data = self._body[idx][1]
+        trivia = getattr(old_data, "trivia", None)
+        if trivia and idx > 0 and getattr(trivia, "comment", None):
+            self._body[idx - 1][1].comment(trivia.comment)
         self._body[idx] = (None, Null())
```

Comment:
It is better to leave the comment at its own line as a line comment:
```python
self._body[idx] = (None, Comment(Trivia(comment_ws="", comment=trivia.comment)))
```

Reply:
I would do that but we have a regression test that makes sure we move it one line up

Comments in toml spec need to belong to another element as far as I understood, right?

Did you write the test for this? https://github.com/pypa/pipenv/blob/master/tests/integration/test_uninstall.py#L154

Reply:
Yes, but the test only checks whether the comment is preserved. In fact it is leaved as a line comment(That is also what old `contoml` does). TOML spec allows [line-level comments](https://github.com/toml-lang/toml#user-content-example), too.

Reply:
ah perfect, that makes so much more sense. Doesn't this level of context-aware parsing make it much slower to parse though?